server/chat-plugins/wifi.ts

```diff
@@ -437,13 +437,13 @@ export class LotteryGiveaway extends Giveaway {
 		this.clearTimer();
 
 		const userlist = Object.values(this.joined);
-		if (userlist.length < this.maxWinners) {
+		if (userlist.length === 0) {
 			this.changeUhtml('<p style="text-align:center;font-size:13pt;font-weight:bold;">The giveaway was forcibly ended.</p>');
 			delete this.room.giveaway;
-			return this.room.send("The giveaway has been forcibly ended as there are not enough participants.");
+			return this.room.send("The giveaway has been forcibly ended as there are no participants.");
 		}
 
-		while (this.winners.length < this.maxWinners) {
+		while (this.winners.length < this.maxWinners && this.userlist > 0) {
```

Comment:
```suggestion
		while (this.winners.length < this.maxWinners && userlist.length > 0) {
```

Reply:
Never mind, looks like you figured it out :)

Reply:
Yes, I hadn't set the project up locally to run the tests myself.
Seeing the logical issue, I set the repository up locally, fixed this issue,
and have tested, rebased and re-submitted.

Reply:
Testing locally is always a good plan.

Reply:
I thought I already told you this, but I ran the checks locally and they're passing.